### PR TITLE
Divider: updated design guidelines

### PR DESCRIPTION
**Pull request by AlbertCarreras**
### Summary

Divider: updated design guidelines

### Links

- [Jira](link to Jira ticket(s))
- [TDD](link to Paper doc)
- [Figma](link to Figma file)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/pages/divider.js [hunk 1]
```diff
@@ -1,64 +1,309 @@
...
+import Page from '../components/Page.js';
 
-const cards: Array<Node> = [];
-const card = (c) => cards.push(c);
+export default function DividerPage(): Node {
+  return (
+    <Page title="Divider">
+      <PageHeader
+        name="Divider"
+        description="If you have two things that need to be separated, put a `Divider` between them."
```
Comment (dangerismycat) [inline docs/pages/divider.js:12]: - "things" is a bit generic…perhaps "elements"?
- No need for the backticks around Divider — we've moved away from that style for component names

File: docs/pages/divider.js [hunk 2]
```diff
@@ -1,64 +1,309 @@
...
-card(
-  <PageHeader
-    name="Divider"
-    description="If you have two things that need to be separated, put a `Divider` between them."
-  />,
-);
+  return (
+    <Flex>
+      <Box paddingX={4}>
+        <Block title="Discover ideas" text="Use the searh bar to discover ideas, people and trends. Explore suggested topics or search for topics of your own." />
```
Comment (dangerismycat) [inline docs/pages/divider.js:25]: typo: s/'searh'/'search'

File: docs/pages/divider.js [hunk 3]
```diff
@@ -1,64 +1,309 @@
...
+      </Box>
+      <Divider />
+      <Box paddingX={4}>
+        <Block title="Create Pins" text="Upload an image from your computer or mobile device to create a Pin. You can also create Pins from images yu find online." />
```
Comment (dangerismycat) [inline docs/pages/divider.js:29]: typo: s/'yu'/'you'

File: docs/pages/divider.js [hunk 4]
```diff
@@ -1,64 +1,309 @@
...
+    </Flex>
+  );
+}
+`}
+      />
+      <MainSection name="Usage guidelines">
+        <MainSection.Subsection>
+          <MainSection.Card
+            description={`
+Dividers can be a subtle way to separate content, however, white space can be used to accomplish the same goal. Use dividers sparingly to avoid creating unnecessary clutter. Dividers should be used mainly to separate groups of related content.
```
Comment (dangerismycat) [inline docs/pages/divider.js:40]: The first sentence is a bit of a run-on. Perhaps a semi-colon or period between "content" and "however"?

File: docs/pages/divider.js [hunk 5]
```diff
@@ -1,64 +1,309 @@
...
+    <Flex direction="column" gap={2}>
+      <Heading size="sm">{title}</Heading>
+      <Text size="md">{text}</Text>
+    </Flex>
+  );
 
-card(<PropTable props={[]} />);
+  return (
+    <Flex direction="column" gap={10}>
+      <Block title="Discover ideas" text="Use the searh bar to discover ideas, people and trends. Explore suggested topics or search for topics of your own." />
```
Comment (dangerismycat) [inline docs/pages/divider.js:62]: same typos in these blocks in the various examples

File: docs/pages/divider.js [hunk 6]
```diff
@@ -1,64 +1,309 @@
...
+  const Block = ({ title, text }) => (
+    <Flex direction="column" gap={2}>
+      <Heading size="sm">{title}</Heading>
+      <Text size="md">{text}</Text>
+    </Flex>
+  );
 
-card(<PropTable props={[]} />);
+  return (
+    <Flex direction="column" gap={10}>
```
Comment (dangerismycat) [inline docs/pages/divider.js:61]: Can we mix it up a bit with these examples? Seeing the same copy so many times in a row is a bit boring and doesn't engage the reader.

File: docs/pages/divider.js [hunk 7]
```diff
@@ -1,64 +1,309 @@
...
+        `}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection name="Best practices">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            description="Use whitespace primarily to separate groups of related content"
```
Comment (dangerismycat) [inline docs/pages/divider.js:50]: These should end in a period (this example and several others)

File: docs/pages/divider.js [hunk 8]
```diff
@@ -1,64 +1,309 @@
...
+}
+`}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection name="Accessibility">
+        <MainSection.Subsection description="The Divider is not focusable and is treated as decorative. Screen readers on tab navigation don't announce Dividers but do announce them on left/right quick navigation." />
+      </MainSection>
+      <MainSection name="Localization">
+        <MainSection.Subsection description="If you are aligning the Divider to content, ensure that it switches sides according to the contents start-alignment" />
```
Comment (dangerismycat) [inline docs/pages/divider.js:249]: "…according to the content's start-end alignment."
- apostrophe on "content"
- "start-end" to clarify which alignment we're referring to
- period at the end

File: docs/pages/divider.js [hunk 9]
```diff
@@ -1,64 +1,309 @@
...
+      <Divider />
+      <Box paddingX={4}>
+        <Block title="Create Pins" text="Upload an image from your computer or mobile device to create a Pin. You can also create Pins from images yu find online." />
+      </Box>
+    </Flex>
+  );
+}
+`}
+      />
+      <MainSection name="Usage guidelines">
```
Comment (dangerismycat) [inline docs/pages/divider.js:36]: Can this use our standard "Do/Don't" format?

File: docs/pages/divider.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js:12]: DONE

File: docs/pages/divider.js (hunk 5 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js:62]: DONE

File: docs/pages/divider.js (hunk 4 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js:40]: DONE

File: docs/pages/divider.js (hunk 7 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js:50]: DONE

File: docs/pages/divider.js (hunk 9 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js:36]: DONE

File: docs/pages/divider.js (hunk 3 above)
Comment (dangerismycat) [reply docs/pages/divider.js:29]: I'm still seeing this typo (here and where the example is repeated below)

File: docs/pages/divider.js (hunk 9 above)
Comment (dangerismycat) [reply docs/pages/divider.js:36]: This still needs a "Don't", and part of this "Do" is referring to negative guidance. Can this be split into a "Do"/"Don't" pair?

File: docs/pages/divider.js [hunk 10]
```diff
@@ -1,64 +1,388 @@
...
+      <Block title="Plants" url="https://i.ibb.co/FY2MKr5/stock6.jpg" />
+    </Flex>
+  );
+}
+`}
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Inset the Divider in a way that causes it to be free-floating or separated from content."
```
Comment (dangerismycat) [inline docs/pages/divider.js:205]: nit: no article needed, just "Inset Divider in a way…"

File: docs/pages/divider.js [hunk 11]
```diff
@@ -1,64 +1,388 @@
...
+            cardSize="md"
+            type="do"
+            description="Use Dividers to separate collections of content or create the appearance of containers."
+            defaultCode={`
+function Example() {
+  return (
+    <Flex width="100%" height="100%">
+      <Box paddingX={4}>
+        <Flex direction="column" gap={2} width={120}>
+        {["Public profile", "Account settings", "Home feed tuner", "Claim", "Permissions", "Notifications"].map((item, idx) => <Text underline={idx === 0}weight="bold" key={idx} size="md" wrap="none">{item}</Text>)}
```
Comment (dangerismycat) [inline docs/pages/divider.js:271]: The underline here is a bit odd…I know it's supposed to indicate the currently-selected item, but it reads as a link, so it's unexpected that it's not a link. Perhaps just remove the underline? Or implement using a Box similar to how it's implemented in Pinboard.

File: docs/pages/divider.js [hunk 12]
```diff
@@ -1,64 +1,388 @@
...
+      </Box>
+    </Flex>
+  );
+}
+`}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection name="Accessibility">
+        <MainSection.Subsection description="The Divider is not focusable and is treated as decorative. Screen readers on tab navigation don't announce Dividers but do announce them on left/right quick navigation." />
```
Comment (dangerismycat) [inline docs/pages/divider.js:325]: nit: no article needed, just "Divider is not focusable…"

File: docs/pages/divider.js [hunk 13]
```diff
@@ -1,64 +1,388 @@
...
+      </MainSection>
+      <MainSection name="Localization">
+        <MainSection.Subsection description="If you are aligning the Divider to content, ensure that it switches sides according to the content's start-end alignment." />
```
Comment (dangerismycat) [inline docs/pages/divider.js:328]: same nit about no article

File: docs/pages/divider.js [hunk 14]
```diff
@@ -1,64 +1,388 @@
...
+      </MainSection>
+      <MainSection name="Variants">
+        <MainSection.Subsection
+          title="Orientation"
+          columns={2}
+          description="You can use this component for a visual divider between two elements. Placing it within a [Flex](/flex) layout with a direction of `row` will cause the Divider to become vertical."
```
Comment (dangerismycat) [inline docs/pages/divider.js:334]: The copy seems a little off…perhaps something like:
"You can use this component as a vertical divider between two elements. Placing it within a Flex layout with a direction of `"row"` will shift Divider to a vertical orientation."

File: docs/pages/divider.js (hunk 10 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js:205]: DONE

File: docs/pages/divider.js (hunk 12 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js:325]: DONE

File: docs/pages/divider.js (hunk 13 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js:328]: DONE

File: docs/pages/divider.js (hunk 14 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js:334]: Done

File: docs/pages/divider.js (hunk 9 above)
Comment (dangerismycat) [reply docs/pages/divider.js:36]: These are now just repeating the first Best Practices pair. Perhaps work with Kate to develop content that captures the original Usage Guidelines without repeating later content?

File: docs/pages/divider.js [hunk 15]
```diff
@@ -1,64 +1,387 @@
...
+    </Flex>
+  );
+}
+`}
+      />
+      <MainSection name="Usage guidelines">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            type="do"
+            description="- Use Dividers or whitespace subt to seperate groups of related content."
```
Comment (dangerismycat) [inline docs/pages/divider.js:40]: I'm not sure what "subt" means here

File: docs/pages/divider.js [hunk 16]
```diff
@@ -1,64 +1,387 @@
...
+          />
+          <MainSection.Card
+            type="don't"
+            description="- Overuse Dividers as this can lead to uncessary clutter on the page."
```
Comment (dangerismycat) [inline docs/pages/divider.js:44]: typo: "unnecessary"

File: docs/pages/divider.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js]: DONE

File: docs/pages/divider.js (hunk 9 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js:36]: DISCUSSED, CHANGED

File: docs/pages/divider.js (hunk 11 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js]: DONE, changed to Tabs

File: docs/pages/divider.js (hunk 16 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js]: DONE

File: docs/pages/divider.js (hunk 15 above)
Comment (AlbertCarreras) [reply docs/pages/divider.js]: DONE

File: docs/pages/divider.js [hunk 17]
```diff
@@ -1,64 +1,385 @@
...
+import Page from '../components/Page.js';
 
-const cards: Array<Node> = [];
-const card = (c) => cards.push(c);
+export default function DividerPage(): Node {
+  return (
+    <Page title="Divider">
+      <PageHeader
+        name="Divider"
+        description="A light gray 1px horizontal or vertical line which groups and divides content in lists and layouts."
```
Comment (dangerismycat) [inline docs/pages/divider.js:12]: ah, missed this previously: this should live as a comment in the component file and get used here as the generated version (`generatedDocGen?.description`), so users see the description on hover in their IDE

File: docs/pages/divider.js (hunk 9 above)
Comment (dangerismycat) [reply docs/pages/divider.js:36]: nit: if we're not pairing it with a "Don't", I think we can forgo the "Do". When not in counterpoint to anything, the "Do" is superfluous

**Review by dangerismycat — APPROVED** [review]

**Pull request merged**